src/app/index.tsx

```diff
@@ -20,7 +21,7 @@ const App: React.FunctionComponent = () => (
     <PollingContextProvider>
       <LocalStorageContextProvider>
         <NetworkContextProvider>
-          <Router>
+          <Router getUserConfirmation={noop}>
```

Comment:
This seems odd, do we need this to return a default message if/when a native alert is used? (see my general comment)

Reply:
Nevermind, I see that this prevents history.block from defaulting to a native confirm dialog.

Reply:
Yea took me a few to grok what was happening here. Turns out whatever the callback to block() returns is provided here so I suppose if you need to do further processing to determine if the route should be blocked or not, you have the ability to do that in a more global manner. I set up an enum that we can use to control the passage - but it's not required. By setting a function here, we're overriding the default behavior of react-router block implementation, so we need to at least set something, even if it's just a noop like it was before. I left it more verbose just for discussion sake so people could see it in action and in case it ties in with preventing the tab close as you mentioned above.